### PR TITLE
Submission

### PR DESCRIPTION
**Pull request by mantasbu**
Exercise submission with incomplete client side

### CODE REVIEW

File: InsiteTeamTask/Repositories/AttendanceRepo/AttendanceRepository.cs [hunk 1]
```diff
@@ -0,0 +1,142 @@
...
+            {
+                foreach (Member member in members)
+                {
+                    if (member.ProductId == product.Id)
+                    {
+                        attendanceList.Add(new Attendance()
+                        {
+                            Barcode = "N/A",
+                            MemberId = member.Id
+                        });
```
Comment (FortressInsite) [inline InsiteTeamTask/Repositories/AttendanceRepo/AttendanceRepository.cs:64]: There is almost *always* a better solution than using nested loops, especially for a process of this simplicity. Here you would have been better off using a combination of `Any()` and `Where()` to only find `Members` with a matching `ProductId`, and then creating `Attendance` objects as per my last comment:
```
var validMembers = members.Where(m => products.Any(p => p.Id == m.ProductId));
var memberAttendances = validMembers.Select(m => new Attendance() { MemberId = m.Id });
```

File: InsiteTeamTask/Repositories/AttendanceRepo/AttendanceRepository.cs [hunk 2]
```diff
@@ -0,0 +1,142 @@
...
+                    }
+                }
+            }
+
+
+            return attendanceList;
+        }
+
+        public List<Attendance> GetAttendanceListByGameId(int gameId)
```
Comment (FortressInsite) [inline InsiteTeamTask/Repositories/AttendanceRepo/AttendanceRepository.cs:88]: I appreciate your thoroughness, but you've done more work than you need to by creating these methods. Game 3 in one season may be completely different to game 3 in another, so you're never going to want to find attendance by game without also limiting by season.

File: InsiteTeamTask/Repositories/AttendanceRepo/AttendanceRepository.cs [hunk 3]
```diff
@@ -0,0 +1,142 @@
...
+
+            return attendanceList;
+        }
+
+        public List<Attendance> GetAttendanceListBySeasonIdAndGameId(int seasonId, int gameId)
+        {
+            List<Ticket> tickets = service.Tickets().ToList();
+            List<Product> products = service.Products().Where(x => x.SeasonId == seasonId)
+                                                       .Where(x => x.GameId == gameId)
+                                                       .ToList();
```
Comment (FortressInsite) [inline InsiteTeamTask/Repositories/AttendanceRepo/AttendanceRepository.cs:120]: The collection of Products includes individual tickets (`Type = ProductType.Ticket`) with access to one game, and season passes (`Type = ProductType.Member`) with access to all games in a season, meaning they have access to the game specified here even though their GameId doesn't match (GameId is zero on all member products). Therefore, to get the right set of products you needed to do something like this:
```
var ticketProducts = service.Products().Where(p => p.SeasonId == seasonId && p.GameId == gameId)
var memberProducts = service.Products().Where(p => p.SeasonId == seasonId && p.Type == ProductType.Member);
```
Also, your `Where()` methods can be joined as I have shown above

File: InsiteTeamTask/Repositories/EventRepo/EventRepository.cs [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+        private readonly IDataService _mockData;
+
+        public EventRepository()
+        {
+            _mockData = new MockDataService();
+        }
+
+        public Event Get(int id)
+        {
+            return _mockData.Events().Where(x => x.Id == id).FirstOrDefault();
```
Comment (FortressInsite) [inline InsiteTeamTask/Repositories/EventRepo/EventRepository.cs:19]: Your lambda method can be used as an argument for the `FirstOrDefault()` method, meaning the `Where()` isn't needed:
```
return _mockData.Events().FirstOrDefault(e => e.Id == id);
```

File: InsiteTeamTask/Repositories/EventRepo/EventRepository.cs [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+        }
+
+        public List<Event> GetAll()
```
Comment (FortressInsite) [inline InsiteTeamTask/Repositories/EventRepo/EventRepository.cs:22]: Changing this return type to an `IEnumerable` would keep things nice and generic, and means that this method doesn't have to do anything except pass between the calling method and the data service

**Comment by FortressInsite** [comment]
I'm not going to pick through your Angular code as it hasn't been finished, but from what I can see the work you've done there sets a good foundation for the rest of the front end to be built upon. All you needed was to inject your services into your components and use them to fetch some data.

For the back-end, I've left a few comments but there's nothing here that worries me, just things I think you could improve upon (use of LINQ, return types, smaller stuff). Your code was very easy to read and review, so good work.

**Comment by mantasbu** [comment]
Thank you so much for your feedback

2020-10-09, pn, 13:21 FortressInsite <notifications@github.com> rašė:

**Pull request closed without merging**